pkg/providers/pricing/pricing.go

```diff
@@ -209,6 +209,9 @@ func (p *Provider) UpdateOnDemandPricing(ctx context.Context) error {
 
 	p.onDemandPrices = lo.Assign(onDemandPrices, onDemandMetalPrices)
 	p.onDemandUpdateTime = time.Now()
+	for instanceType, price := range p.onDemandPrices {
+		instanceEstimate.WithLabelValues(instanceType, ec2.UsageClassTypeOnDemand, p.region, "").Set(price)
```

Comment:
nit: Using `WithLabels` makes it a bit more straightforward to see what each thing maps to when the labels are generated